internal/xpkg/name.go

```diff
@@ -65,7 +65,7 @@ func FriendlyID(name, hash string) string {
 }
 
 // ToDNSLabel converts the string to a valid DNS label.
-func ToDNSLabel(s string) string { // nolint:gocyclo
+func ToDNSLabel(s string) string { //nolint:gocyclo // TODO(negz): Document the conditions in this function.
```

Comment:
nit: the `TODO` here does not seem to be answering the reasoning linter exception :)